### ISSUE TITLE
Unexpected Model Download Behaviour in YOLOv8

### ISSUE THREAD

**Issue by didik12ax** [ultralytics/ultralytics#4309]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello YOLOv8 community,

I recently tried to use the yolov8m.pt model, but during the initialization, I encountered an unexpected behaviour. The system mentioned: AMP: running Automatic mixed precision (AMP) checks with yolov8n... and proceeded to download the yolov8n.pt model. This was not the expected behaviour, as I intended to use only the yolov8m.pt model.

Here's a snippet from the logs I received:

`Model summary: 295 layers, 25857478 parameters, 25857462 gradients
Transferred 469/475 items from pretrained weights
AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
Downloading https://github.com/ultralytics/assets/releases/download/v0.0.0/yolov8n.pt to 'yolov8n.pt'...
100%|█████████████████████████████████████████████████████████████████████████████| 6.23M/6.23M [00:00<00:00, 10.6MB/s]
AMP: checks passed
train: Scanning E:\Diki\DHB\dataset54\train\labels.cache... 1248 images, 0 backgrounds, 0 corrupt: 100%|██████████| 124
val: Scanning E:\Diki\DHB\dataset54\valid\labels.cache... 121 images, 0 backgrounds, 0 corrupt: 100%|██████████| 121/12
Plotting labels to runs\detect\train4\labels.jpg...
optimizer: AdamW(lr=0.001667, momentum=0.9) with parameter groups 77 weight(decay=0.0), 84 weight(decay=0.00046875), 83 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 8 dataloader workers
Logging results to runs\detect\train4
Starting training for 100 epochs...`

I'm trying to understand the reasoning behind this behavior. Why would the initialization of one model trigger checks and potential downloads for a different model? Is there any way to prevent this automatic download?

**Comment by glenn-jocher** [comment]
@didik12ax hello,

Thank you for bringing the unexpected behaviour to our attention. The behavior you're experiencing appears to be the result of the Automatic Mixed Precision (AMP) checks that YOLOv8 runs for all its models.

When initializing, YOLOv8 performs certain sanity checks to ensure that the optimization process will run smoothly. This includes automatic mixed precision tests using a particular intermediate model (YOLOv8n in your case). During these tests, it uses this model as a reference to determine optimal settings for increased training speed and efficiency for your specified model (yolov8m.pt in your case). 

If the reference model isn't available locally, the software automatically downloads it, hence the behavior you observed. 

At this point in time, there's not a system in place within the YOLOv8 design to disable this automatic download as the AMP checks are integral to delivering optimal training results. 

We understand how this might cause confusion and will look into making this process clearer in future updates. Thanks for contributing to the YOLOv8 community!

**Comment by didik12ax** [comment]
Hi @glenn-jocher, 

First, I'd like to express my appreciation for the detailed response regarding the AMP checks and the automatic download of the yolov8n.pt model. I now understand the reasoning behind this behavior.

Following up on that, I'm eager to test and understand the performance of the yolov8m.pt model. To ensure a smooth execution, could you provide guidance or recommendations on the optimal environment setup to run the yolov8m model? Specifically:

Are there any specific software dependencies or versions that work best with yolov8m?
Would you recommend any particular hardware configurations, especially regarding GPU specifications, to maximize the performance of this model?
Are there any known issues or considerations to be aware of when working with yolov8m?
Your insights will be immensely helpful as I aim to explore the full potential of the yolov8m model.

Thank you in advance for your time and support!
